### PR TITLE
Add 'try' parameter to H5Z_find, and remove calls to H5E_clear_stack()

### PR DESCRIPTION
**Pull request by qkoziol**

### CODE REVIEW

**Comment by qkoziol** [comment]
And pause & resume pushing errors on the default error stack when calling user's filter routine

**Review by derobins — APPROVED** [review]

**Comment by qkoziol** [comment]
*bump*  :-) 

Anyone free to review this small set of changes?

File: src/H5Opline.c [hunk 1]
```diff
@@ -302,15 +302,18 @@ H5O__pline_encode(H5F_t H5_ATTR_UNUSED *f, uint8_t *p /*out*/, const void *mesg)
...
              * Get the filter name.  If the pipeline message has a name in it then
              * use that one.  Otherwise try to look up the filter and get the name
              * as it was registered.
              */
-            if (NULL == (name = filter->name) && (cls = H5Z_find(filter->id)))
-                name = cls->name;
+            if (NULL == (name = filter->name)) {
+                H5Z_class2_t *cls; /* Filter class */
+
+                H5Z_find(true, filter->id, &cls);
```
Comment (bmribler) [inline src/H5Opline.c:313]: No checking returned value?

**Review by mattjala — APPROVED** [review]

File: src/H5Opline.c (hunk 1 above)
Comment (qkoziol) [reply src/H5Opline.c:313]: Correct - it's a "speculative" lookup ('try' parameter set to true), which doesn't push errors

**Pull request merged**